controllers/cloud_config_sync_controller.go

```diff
@@ -0,0 +1,158 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	openshiftConfigNamespace        = "openshift-config"
+	openshiftManagedConfigNamespace = "openshift-config-managed"
+	managedCloudConfigMapName       = "kube-cloud-config"
+
+	cloudConfigMapName = "cloud-conf"
+	defaultConfigKey   = "cloud.conf"
+)
+
+type CloudConfigReconciler struct {
+	client.Client
+	Scheme          *runtime.Scheme
+	Recorder        record.EventRecorder
+	TargetNamespace string
+}
+
+func (r *CloudConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.Infof("%s emitted event, syncing cloud-conf ConfigMap", req)
+
+	// Use kube-cloud-config from openshift-config-managed namespace as default source.
+	// If it is not exists try to use cloud-config reference from infra resource.
+	// https://github.com/openshift/library-go/blob/master/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L82
+	defaultSourceCMObjectKey := client.ObjectKey{
+		Name: managedCloudConfigMapName, Namespace: openshiftManagedConfigNamespace,
+	}
+	sourceCM := &corev1.ConfigMap{}
+	if err := r.Get(ctx, defaultSourceCMObjectKey, sourceCM); errors.IsNotFound(err) {
+		klog.Warningf("managed cloud-config is not found trying to get infra one")
+		infra := &configv1.Infrastructure{}
+		if err := r.Get(ctx, client.ObjectKey{Name: infrastructureResourceName}, infra); err != nil {
+			klog.Errorf("infrastructure resource not found")
+			return ctrl.Result{}, err
+		}
+
+		openshiftUnmanagedCMKey := client.ObjectKey{Name: infra.Spec.CloudConfig.Name, Namespace: openshiftConfigNamespace}
+		if err := r.Get(ctx, openshiftUnmanagedCMKey, sourceCM); err != nil {
+			klog.Errorf("unable to get cloud-config for sync")
+			return ctrl.Result{}, err
+		}
+		sourceCM, err = r.prepareSourceConfigMap(sourceCM, infra)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else if err != nil {
+		klog.Errorf("unable to get managed cloud-config for sync")
+		return ctrl.Result{}, err
+	}
+
+	targetCM := &corev1.ConfigMap{}
+	targetConfigMapKey := client.ObjectKey{
+		Namespace: r.TargetNamespace,
+		Name:      cloudConfigMapName,
+	}
+
+	if err := r.Get(ctx, targetConfigMapKey, targetCM); err != nil && !errors.IsNotFound(err) {
```

Comment:
```suggestion
	// If the config does not exist, it will be created later, so we can ignore a Not Found error
	if err := r.Get(ctx, targetConfigMapKey, targetCM); err != nil && !errors.IsNotFound(err) {
```